projects/step-frontend/src/lib/modules/execution/components/schedule-overview/cross-execution-dashboard/heatmap/legend/gradient-legend.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, computed, input, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { HeatMapColor } from '../types/heatmap-types';
+
+interface ComputedStop {
+  color: string;
+  label: string;
+  posPct: number; // 0..100
+}
+
+@Component({
+  selector: 'step-heatmap-gradient-legend',
+  templateUrl: './gradient-legend.component.html',
+  styleUrls: ['./gradient-legend.component.scss'],
+  standalone: false,
+})
+export class GradientLegendComponent {
+  gradientStops = input.required<HeatMapColor[]>();
+  barHeight = input<number>(14);
```

Comment:
Field modificators are missing.
Inputs should be readonly.
Fields that are used inside templates should be protected.
All other fields should be private.